src/spentindex.h

```diff
@@ -53,11 +54,19 @@ struct CSpentIndexValue {
         READWRITE(inputIndex);
         READWRITE(blockHeight);
         READWRITE(satoshis);
-        READWRITE(addressType);
+
+        //int is used for backward compatibility
+        int addressTypeInt = 0;
+        if (!ser_action.ForRead()) //serialize
+            addressTypeInt = (int)addressType;
+        READWRITE(addressTypeInt);
+        if (ser_action.ForRead()) //unserialize
+            addressType = (AddressType)addressTypeInt;
+
```

Comment:
Do you think this code would work properly also if we remove the distinction between serialization and deserialization?

```suggestion

        // int is used for backward compatibility
        int addressTypeInt = static_cast<int>(addressType);
        READWRITE(addressTypeInt);
        addressType = static_cast<AddressType>(addressTypeInt);

```

Reply:
Done

Reply:
Done